TSTutorial/homeworks/hw_4/Pizzeria.ts

```diff
@@ -0,0 +1,49 @@
+import { TOrder } from "./Types";
+import {OrderStatusEnum} from "./Enums";
+import { OrderService } from "./OrderService";
+
+export default class Pizzeria {
+    private orderService: OrderService
+    constructor(private name: string, private adress: string, private workingHours: string) {
+        console.log(`Welcome to Pizzeria '${this.name}' on adress '${this.adress}' at '${this.workingHours}'`)
+        this.orderService = new OrderService()
+    }
+
+    addOrders(order: TOrder[]): void {
+        this.orderService.addOrder(order)
+        if (this.orderService.getOrderByStatus(OrderStatusEnum.PROCESSING) !== null
+            && this.orderService.getOrderByStatus(OrderStatusEnum.IN_PROGRESS) === null) {
+            this.orderService.setStatusToOrder(OrderStatusEnum.PROCESSING, OrderStatusEnum.IN_PROGRESS)
+        } else this.orderService.setStatusToOrder(OrderStatusEnum.PROCESSING, OrderStatusEnum.PENDING)
```

Comment:
This implementation should be a private part of OrderService class. Pizzeria class should not have any knowledge about how it works under cover, and must rely only on the provided inverface